### PR TITLE
ci: add verdaccio to pipeline

### PR DESCRIPTION
**Pull request by iartemiev**
* Utilize Verdaccio for installing new source code into integration tests (instead of linking packages)
* Refactor test jobs, make them DRY by abstracting logic into parameterized commands
* Streamline workflow

Need to merge this samples PR before this gets merged: https://github.com/aws-amplify/amplify-js-samples-staging/pull/93

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -537,124 +383,76 @@ jobs:
...
-                - ui-components/master
-                - 1.0-stable
+            <<: *releasable_branches
           requires:
             - unit_test
-            - integ_react_predictions
+            # - integ_react_predictions
             - integ_react_auth
-            - integ_angular_auth
+            # - integ_angular_auth
```
Comment (sammartinez) [inline .circleci/config.yml:453]: What is the reason for turning off these test?

Comment (iartemiev) [reply .circleci/config.yml:453]: Just turning them off until the linked samples PR is merged

File: .circleci/config.yml [hunk 2]
```diff
@@ -34,11 +42,163 @@ commands:
...
+      - run:
+          name: 'Install << parameters.test_name >> sample'
+          command: |
+            echo "Current NPM registry: " $(yarn config get registry)
+            yarn
+      - run:
+          name: 'Run << parameters.test_name >> UI tests'
+          command: |
+            cd ~/amplify-js-samples-staging-ui
+            yarn ci:test << parameters.framework >> << parameters.spec >>
```
Comment (jordanranz) [inline .circleci/config.yml:132]: I believe this is missing a category. Should be:
```
ci:test  << parameters.framework >> << parameters.category >> << parameters.sample_name >> << parameters.spec >>
```
If `spec` is omitted, it will use the `sample_name` as the `spec` name

File: .circleci/config.yml [hunk 3]
```diff
@@ -97,182 +259,49 @@ jobs:
...
+          framework: vue
+          spec: auth/authenticator.spec.js
+
+  integ_react_predictions:
+    executor: js-test-executor
+    <<: *test_env_vars
+    working_directory: ~/amplify-js-samples-staging/samples/react/predictions/multi-user-translation
+    steps:
+      - integ_test_js:
+          test_name: 'Vue Predictions'
```
Comment (jordanranz) [inline .circleci/config.yml:299]: Is this Vue or React?

File: packages/aws-amplify-react/__tests__/__snapshots__/AmplifyUI-test.tsx.snap [hunk 4]
```diff
@@ -205,7 +205,9 @@ exports[`AmplifyUi test render SectionFooter correctly 1`] = `
   className="amplify-section-footer"
   style={Object {}}
 >
-  <Component />
+  <Component
```
Comment (jordanranz) [inline packages/aws-amplify-react/__tests__/__snapshots__/AmplifyUI-test.tsx.snap:208]: This should probably be put into a different PR. (It actually may already have)

File: .circleci/config.yml (hunk 2 above)
Comment (iartemiev) [reply .circleci/config.yml:132]: For the UI tests, I'm passing the category name in the spec param:
```
  integ_react_auth_ui:
    executor: js-test-executor
    <<: *test_env_vars
    working_directory: ~/amplify-js-samples-staging-ui/samples/react/auth/amplify-authenticator
    steps:
      - integ_test_ui:
          test_name: 'React Authenticator'
          framework: react
          spec: 'auth amplify-authenticator'
```
We can split it out into a separate param though, if you think that would be better

File: .circleci/config.yml (hunk 3 above)
Comment (iartemiev) [reply .circleci/config.yml:299]: Should be React. Fixed

File: packages/aws-amplify-react/__tests__/__snapshots__/AmplifyUI-test.tsx.snap (hunk 4 above)
Comment (iartemiev) [reply packages/aws-amplify-react/__tests__/__snapshots__/AmplifyUI-test.tsx.snap:208]: Yep, we've already merged in an updated snap into master, I'll remove this one

Comment (iartemiev) [reply packages/aws-amplify-react/__tests__/__snapshots__/AmplifyUI-test.tsx.snap]: The reason I had added it here is because I couldn't do a push without updating the snap, since the tests were failing

File: .circleci/config.yml (hunk 2 above)
Comment (jordanranz) [reply .circleci/config.yml:132]: Yea, I think splitting it may be simpler. Keep in mind my comment about a sample name and spec name possibly being different.

Comment (iartemiev) [reply .circleci/config.yml]: @jordanranz good call. I hadn't scrolled all the way to the right. I've added them both

File: .circleci/config.yml [hunk 5]
```diff
@@ -537,124 +396,115 @@ jobs:
...
+            - amplify-ssh-deps
+      - run: yarn && yarn build
+      - run:
+          name: 'Post-release steps'
+          command: |
+            git config --global user.email $GITHUB_EMAIL
+            git config --global user.name $GITHUB_USER
+            git pull origin release
+            yarn && yarn build
+            git add ./packages/core/src/Platform/version.ts
```
Comment (sammartinez) [inline .circleci/config.yml:417]: Will this be needed for each no matter what at this point?

Comment (iartemiev) [reply .circleci/config.yml:417]: Yep, for each stable release
